src/Middleware/RateLimiter.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Envato\Middleware;
+
+class RateLimiter {
+  /**
+   * @var string Filename of the lock file that is used to hold the unix
+   * timestamp of rate limit timeouts.
+   */
+  public $rate_limit_file_name = 'envato_api_sdk.lock';
+
+  /**
+   * Checks if the client is currently throttled from making requests.
+   *
+   * @return boolean Whether the client is prohibited from initiating new
+   * connections.
+   */
+  public function isThrottled() {
+    $rate_limit_timeout_value = $this->rateLimitTimeoutValue();
+
+    if (time() > $rate_limit_timeout_value) {
+      $this->removeThrottle();
+      return FALSE;
+    } else {
+      return TRUE;
+    }
+  }
+
+  /**
+   * Create a rate limit flag.
+   *
+   * @param int $retry_after_value Time in seconds that the client has been rate
+   * limited for.
+   */
+  public function setThrottle($retry_after_value) {
+    $retry_after_timestamp = time() + $retry_after_value;
+
+    if (!$this->temporary_directory_writable()) {
+      return FALSE;
+    }
+
+    $lock_file = fopen($this->full_system_path_to_lock_file(), 'w');
```

Comment:
One of the issues I'm needing to work around here is the fact that this client needs to be able to handle graceful degradation without the reliance on external systems for the "throttled" flag. The only thing I know that is consistent across all systems is access to PHP's temporary directory. Using the filesystem for this type of flag isn't the greatest solution however it does give us a starting point to work off.

cc @dtbaker for your thoughts on how compatible this would be on a WordPress installation.